lib/internal/modules/esm/get_format.js

```diff
@@ -161,14 +161,14 @@ function getFileProtocolModuleFormat(url, context = { __proto__: null }, ignoreE
       default: { // The user did not pass `--experimental-default-type`.
         // `source` is undefined when this is called from `defaultResolve`;
         // but this gets called again from `defaultLoad`/`defaultLoadSync`.
-        let parsedSource;
-        if (source) {
-          const { stripTypeScriptTypes } = require('internal/modules/helpers');
-          parsedSource = stripTypeScriptTypes(source, url);
-        }
+        // Since experimental-strip-types depends on detect-module, we always return null
+        // if source is undefined.
+        if (!source) { return null; }
+        const { stripTypeScriptTypes, stringify } = require('internal/modules/helpers');
+        const stringifiedSource = stringify(source);
+        const parsedSource = stripTypeScriptTypes(stringifiedSource, fileURLToPath(url));
         const detectedFormat = detectModuleFormat(parsedSource, url);
-        // When source is undefined, default to module-typescript.
-        const format = detectedFormat ? `${detectedFormat}-typescript` : 'module-typescript';
+        const format = `${detectedFormat}-typescript`;
         if (format === 'module-typescript' && foundPackageJson) {
           // This module has a .js extension, a package.json with no `type` field, and ESM syntax.
```

Comment:
nit: This is in `.ts` branch so,

```suggestion
          // This module has a .ts extension, a package.json with no `type` field, and ESM syntax.
```

Reply:
Since CI is extremely flaky and stressed I'm gonna do it in a followup